--- a/src/io/cmdline.cpp	2023-02-01 11:00:00.000000000 +0100
+++ b/src/io/cmdline.cpp	2023-03-29 00:10:19.000000000 +0200
@@ -918,14 +918,6 @@
                         loadbezel = false;
 
                     if (loadbezel) { // alphanum
-                        string s3 = s1.substr(0, (iLen-4));
-                        for (int i = 0; s3[i] != '\0'; i++) {
-                            if (!isalnum(s3[i]) && s3[i] != int('-')
-                                && s3[i] != int('_') && s3[i] != int('/')) {
-                                loadbezel = false;
-                            }
-                        }
-
                         if (loadbezel)
                             video::set_bezel_file(s);
                     }


--- a/src/video/video.cpp	2023-08-08 00:26:36.000000000 +0200
+++ b/src/video/video.cpp	2023-08-08 00:27:51.000000000 +0200
@@ -319,7 +319,7 @@
                 }
 
                 if (!g_bezel_file.empty()) {
-                    string bezelpath = "bezels/" + g_bezel_file;
+                    string bezelpath = g_bezel_file;
                     g_bezel_texture = IMG_LoadTexture(g_renderer, bezelpath.c_str());
 
                     if (!notify) {
