diff --git a/src/video/SDL_video.c b/src/video/SDL_video.c
index 04a1fdf65..476420bf0 100644
--- a/src/video/SDL_video.c
+++ b/src/video/SDL_video.c
@@ -1779,6 +1779,7 @@ SDL_Window *SDL_CreateWindow(const char *title, int x, int y, int w, int h, Uint
             SDL_zero(fullscreen_mode);
             fullscreen_mode.w = w;
             fullscreen_mode.h = h;
+            fullscreen_mode.refresh_rate = 60;
             if (SDL_GetClosestDisplayModeForDisplay(display, &fullscreen_mode, &closest_mode) != NULL) {
                 bounds.w = closest_mode.w;
                 bounds.h = closest_mode.h;
