From a5b4929087df5195491e86f916ea4e87002a1cf2 Mon Sep 17 00:00:00 2001
From: David Barbion <davidb@230ruedubac.fr>
Date: Wed, 24 Jan 2024 23:36:05 +0100
Subject: [PATCH 4/6] sixaxis-remove-not-needed-checks

---
 profiles/input/server.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/profiles/input/server.c b/profiles/input/server.c
index 79cf08a66..91c352d53 100644
--- a/profiles/input/server.c
+++ b/profiles/input/server.c
@@ -121,8 +121,7 @@ static bool dev_is_sixaxis(const bdaddr_t *src, const bdaddr_t *dst)
 	pid = btd_device_get_product(device);
 
 	cp = get_pairing(vid, pid, NULL);
-	if (cp && (cp->type == CABLE_PAIRING_SIXAXIS ||
-					cp->type == CABLE_PAIRING_DS4))
+	if (cp && cp->type == CABLE_PAIRING_DS4)
 		return true;
 
 	return false;
-- 
2.43.0

