--- a/libretro/retro.cpp	2023-10-27 10:36:37.000000000 +0200
+++ b/libretro/retro.cpp	2023-11-13 02:15:28.410145853 +0100
@@ -128,7 +128,7 @@
          log_cb(RETRO_LOG_INFO, "retro_variable empty value for key '%s'\n", vars[i].key);
          continue;
       }
-      log_cb(RETRO_LOG_INFO, "retro_variable (SYSTEM)    { '%s', '%s' }\n", vars[i].key, vars[i].value);
+      // log_cb(RETRO_LOG_INFO, "retro_variable (SYSTEM)    { '%s', '%s' }\n", vars[i].key, vars[i].value);
 
    }
 
