--- a/libretro.cpp	2023-07-21 19:24:38.000000000 +0200
+++ b/libretro.cpp	2023-10-03 01:59:08.092973695 +0200
@@ -188,7 +188,7 @@
 	// End load per-game settings
 	//
 	char bios_path[2048];
-	snprintf(bios_path, sizeof(bios_path),"%s" SLASH "%s", retro_system_directory, "lynxboot.img");
+	snprintf(bios_path, sizeof(bios_path),"%s" SLASH "%s" SLASH "%s", retro_system_directory, "lynx", "lynxboot.img");
 
 	Load(GameFile, bios_path);
 
