--- evmapy-bd65338c236cd30b4f2d7835733ea5d6b108b75d/evmapy/__main__.py	2023-12-03 15:07:37.087277381 +0100
+++ new-evmapy/__main__.py	2023-12-03 19:23:46.405772048 +0100
@@ -120,7 +120,7 @@
         info = evmapy.util.get_app_info()
         logger = initialize_logging(info['name'], args.debug)
         logger.info("%s %s initializing", info['name'], info['version'])
-        logger.info("running as user %s", info['user'].pw_name)
+        logger.info("running as user %s", "root")
         logger.info("using configuration directory %s", info['config_dir'])
         evmapy.multiplexer.Multiplexer().run()
 
--- evmapy-bd65338c236cd30b4f2d7835733ea5d6b108b75d/evmapy/controller.py	2023-12-03 15:07:37.087277381 +0100
+++ new-evmapy/controller.py	2023-12-03 19:23:46.405772048 +0100
@@ -100,7 +100,7 @@
         return send_request(request)
     except (ConnectionRefusedError, FileNotFoundError):
         exit("No %s instance is running as %s" % (
-            info['name'], info['user'].pw_name
+            info['name'], "root"
         ))
     except TimeoutError:
         exit("Timeout waiting for a response from %s" % info['name'])
--- evmapy-bd65338c236cd30b4f2d7835733ea5d6b108b75d/evmapy/multiplexer.py	2023-12-03 15:07:37.087277381 +0100
+++ new-evmapy/multiplexer.py	2023-12-03 19:23:46.405772048 +0100
@@ -69,7 +69,7 @@
         try:
             signal.signal(signal.SIGHUP, signal.SIG_IGN)
             info = evmapy.util.get_app_info()
-            app_with_user = (info['name'], info['user'].pw_name)
+            app_with_user = (info['name'], "root")
             # Create the control socket
             self._controller = evmapy.controller.Controller(self)
             # Try to open /dev/uinput, failing gracefully
