--- vpinball-bbc48a1957f817dcabed8d6fc55f46156cb0ffef/standalone/cmake/CMakeLists_gl-linux-x64.txt	2024-04-08 04:32:13.000000000 +0200
+++ vpinball-bbc48a1957f817dcabed8d6fc55f46156cb0ffef/standalone/cmake/CMakeLists_gl-linux-x64.new	2024-04-09 20:59:36.300988725 +0200
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.25)
+cmake_minimum_required(VERSION 3.22)
 
 option(POST_BUILD_COPY_EXT_LIBS "Copy external libraries to build directory" ON)
 
@@ -699,4 +699,4 @@
       COMMAND cp -a "${CMAKE_SOURCE_DIR}/standalone/linux-x64/external/lib/*.so" "$<TARGET_FILE_DIR:vpinball>"
       COMMAND cp -a "${CMAKE_SOURCE_DIR}/standalone/linux-x64/external/lib/*.so.*" "$<TARGET_FILE_DIR:vpinball>"
    )
-endif()
\ Pas de fin de ligne à la fin du fichier
+endif()
