--- a/libretro.c	2023-10-19 22:20:38.000000000 +0200
+++ b/libretro.c	2023-10-20 01:07:36.664057331 +0200
@@ -55,7 +55,7 @@
 
 void retro_destroybmp(void);
 
-char bios_file_name[16] = {0};
+char bios_file_name[32] = {0};
 
 uint16_t mbmp[TEX_WIDTH * TEX_HEIGHT];
 uint16_t *mbmp_prev = NULL;
@@ -1107,7 +1107,7 @@
       /* Emulated Hardware */
       var.key   = "o2em_bios";
       var.value = NULL;
-      strcpy(bios_file_name, "o2rom.bin");
+      strcpy(bios_file_name, "o2em/o2rom.bin");
 
       if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
       {
@@ -1115,19 +1115,19 @@
           * copy an invalid value */
          if (!strcmp(var.value, "c52.bin"))
          {
-            strcpy(bios_file_name, "c52.bin");
+            strcpy(bios_file_name, "o2em/c52.bin");
             if (auto_region)
                app_data.euro = 1;
          }
          else if (!strcmp(var.value, "g7400.bin"))
          {
-            strcpy(bios_file_name, "g7400.bin");
+            strcpy(bios_file_name, "videopacplus/g7400.bin");
             if (auto_region)
                app_data.euro = 1;
          }
          else if (!strcmp(var.value, "jopac.bin"))
          {
-            strcpy(bios_file_name, "jopac.bin");
+            strcpy(bios_file_name, "videopacplus/jopac.bin");
             if (auto_region)
                app_data.euro = 1;
          }
