--- a/libretro.c	2023-02-21 17:06:15.000000000 +0100
+++ b/libretro.c	2023-10-03 01:20:44.557804649 +0200
@@ -1100,7 +1100,7 @@
       /* Emulated Hardware */
       var.key   = "o2em_bios";
       var.value = NULL;
-      strcpy(bios_file_name, "o2rom.bin");
+      strcpy(bios_file_name, "o2em/o2rom.bin");
 
       if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
       {
@@ -1108,19 +1108,19 @@
           * copy an invalid value */
          if (!strcmp(var.value, "c52.bin"))
          {
-            strcpy(bios_file_name, "c52.bin");
+            strcpy(bios_file_name, "o2em/c52.bin");
             if (auto_region)
                app_data.euro = 1;
          }
          else if (!strcmp(var.value, "g7400.bin"))
          {
-            strcpy(bios_file_name, "g7400.bin");
+            strcpy(bios_file_name, "videopacplus/g7400.bin");
             if (auto_region)
                app_data.euro = 1;
          }
          else if (!strcmp(var.value, "jopac.bin"))
          {
-            strcpy(bios_file_name, "jopac.bin");
+            strcpy(bios_file_name, "videopacplus/jopac.bin");
             if (auto_region)
                app_data.euro = 1;
          }
