diff --git a/gfx/drivers_context/drm_ctx.c b/gfx/drivers_context/drm_ctx.c
index 3ff1fe9035..6ef476c52a 100644
--- a/gfx/drivers_context/drm_ctx.c
+++ b/gfx/drivers_context/drm_ctx.c
@@ -176,6 +176,8 @@ bool gfx_ctx_drm_load_mode(drmModeModeInfoPtr modeInfo)
       modeInfo->flags = timings.interlaced ? DRM_MODE_FLAG_INTERLACE : 0;
       modeInfo->flags |= timings.v_sync_polarity ? DRM_MODE_FLAG_NVSYNC : DRM_MODE_FLAG_PVSYNC;
       modeInfo->flags |= timings.h_sync_polarity ? DRM_MODE_FLAG_NHSYNC : DRM_MODE_FLAG_PHSYNC;
+      modeInfo->flags |= DRM_MODE_FLAG_CSYNC;
+      modeInfo->flags |= timings.h_sync_polarity && timings.v_sync_polarity ? DRM_MODE_FLAG_NCSYNC : 0;
       modeInfo->type = 0;
       snprintf(modeInfo->name, DRM_DISPLAY_MODE_LEN, "CRT_%ux%u_%u",
                modeInfo->hdisplay, modeInfo->vdisplay, modeInfo->vrefresh);
