--- a/mednafen/ss/ss.cpp	2023-05-28 05:53:42.000000000 +0200
+++ b/mednafen/ss/ss.cpp	2023-10-02 23:20:08.201783475 +0200
@@ -932,7 +932,7 @@
 
    {
       char bios_path[4096];
-      snprintf(bios_path, sizeof(bios_path), "%s" RETRO_SLASH "%s", retro_base_directory, bios_filename);
+      snprintf(bios_path, sizeof(bios_path), "%s" RETRO_SLASH "%s" RETRO_SLASH "%s", retro_base_directory, "saturn", bios_filename);
 
       RFILE *BIOSFile = filestream_open(bios_path,
             RETRO_VFS_FILE_ACCESS_READ,
